SilKit/source/util/SignalHandler.cpp

```diff
@@ -188,7 +205,27 @@ class SignalMonitor
     int _signalNumber{-1};
 };
 
-void systemHandler(int sigNum)
+static inline void setSignalAction(int sigNum, __sighandler_t action)
+{
+    // Check current signal handler action to see if it's set to SIGNAL IGNORE
+    struct sigaction oldAction;
+    sigaction(sigNum, NULL, &oldAction);
+    if (oldAction.sa_handler == SIG_IGN)
+    {
+        // A non-job-control shell wants us to ignore this kind of signal
```

Comment:
should we bail out here? if the sa_handler is set to SIG_IGN, our parent process already set it?

```suggestion
        // A non-job-control shell wants us to ignore this kind of signal
        return;
```